### DISCUSSION TITLE
What is the purpose of "warning blocks"?

### DISCUSSION THREAD

**Discussion by ovizii** [KBlixt/subcleaner#54] [Q&A] [answered]
Are those meant for us to read the logs and give feedback? Not quite sure what to do about them.

**Comment by ovizii** [comment]
I've read the [english.conf](https://github.com/KBlixt/subcleaner/blob/master/regex_profiles/default/english.conf) file and it states:

```
# WARNING_REGEX:
# In the WARNING_REGEX section each individual match from each regex gives one warning to the subtitle block.
# Blocks also receive additional warnings if they are adjacent to other blocks that contain ads.
# 1 warning is ignored
# 2 warnings will be print the block as a WARNING in the log.
# 3 warnings or more will remove the entire block
```

And yet, in the logs of a dry-run I see:

```
          |
          |                                         [---------Warning Blocks----------]
          |                                         898
          |                                         01:48:15,940 --> 01:48:22,555
          |                                         Who I am inside?
          |                                         reasons: (similar_content)
          |
          |                                         912
          |                                         01:49:23,380 --> 01:49:27,420
          |                                         Who I am inside?
          |                                         reasons: (nearby_ad, similar_content)
          |
          |                                         924
          |                                         01:50:21,071 --> 01:50:28,030
          |                                         Who I am inside?
          |                                         reasons: (nearby_ad, similar_content)
          |
          |                                         926
          |                                         01:50:40,000 --> 01:50:46,970
          |                                         Who I am inside?
          |                                         reasons: (close_to_end, adjacent_ad, similar_content)
          |                                         [---------------------------------]

```
Shouldn't warning 926 have been removed? I see 3 listed reasons, or what am I missing here?

**Reply by KBlixt** [reply, marked as answer]
Hi, good spot. I actually have a hidden bonus, if the block doesn't match any regex what so ever it gains -1 point. So a block could have 3 warnings on it and not get deleted. But only if none of the warnings was due to a regex.

Maybe I could improve the message here.

The reason why you see the other blocks with only 2 warnings is due to the promoting effect of that warning. All blocks with the same content will all have the same amount of warnings as the block among them with the highest warning.

I hope that answered your question.

**Reply by ovizii** [reply]
Thanks, its slightly clearer now, let me play a bit with your script and see if I get it :-)

**Comment by KBlixt** [comment]
Pretty much what you said, give feedback and improve the regex. It's a way to reduce false positives while still allowing the script to notify users of close calls that have a relatively high chance to be an add.

There is also a sensitive mode where it prints all warnings, not just the ones receiving 2 warnings.
